docs/release-info/migr-mvd-array.md

```diff
@@ -0,0 +1,135 @@
+
+## Query differences between arrays and MVDs
+
+In SQL queries, Druid operates on arrays differently than MVDs.
+A value in an array column is treated as a single array entity (SQL ARRAY), whereas a value in an MVD column is treated as individual strings (SQL VARCHAR).
+
+For example, consider the same value, `['a', 'b', 'c']` ingested into an array column and an MVD column.
+In your query, you want to filter results by comparing some value with `['a', 'b', 'c']`.
+
+* For array columns, Druid only returns the row when an equality filter matches the entire array.  
+For example: `WHERE "array_column" = ARRAY['a', 'b', 'c']`.
+
+* For MVD columns, Druid returns the row when an equality filter matches any value of the MVD.  
+For example, any of the following filters returns the row for the query:  
+`WHERE "mvd_column" = 'a'`  
+`WHERE "mvd_column" = 'b'`  
+`WHERE "mvd_column" = 'c'`
+
+Note this difference between arrays and MVDs when you write queries that involve filtering or grouping.
+
+The following examples highlight a few analogous queries between arrays and MVDs.
+For more information and examples, see [Querying arrays](../querying/arrays.md#querying-arrays) and [Querying multi-value dimensions](../querying/multi-value-dimensions.md#querying-multi-value-dimensions).
+
+### Example: an element in array
+
+Filter rows that have a certain value in the array or MVD.
+
+#### Array
+
+```sql
+SELECT *
+FROM "array_example"
+WHERE ARRAY_CONTAINS(tags, 't3')
+```
+
+#### MVD
+
+```sql
+SELECT *
+FROM "mvd_example"
+WHERE tags = 't3'
+```
+
+### Example: overlap of two arrays
+
+Filter rows for which the array or MVD overlaps a reference array.
+
+#### Array
+
+```sql
+SELECT *
+FROM "array_example"
+WHERE ARRAY_OVERLAP(tags, ARRAY['t1', 't7'])
+```
+
+#### MVD
+
+```sql
+SELECT *
+FROM "mvd_example"
+WHERE MV_OVERLAP(tags, ARRAY['t1', 't7'])
+```
+
+### Example: group by array elements
+
+Group results by individual array elements.
+
+#### Array
+
+```sql
+SELECT label, strings
+FROM "array_example" CROSS JOIN UNNEST(tags) as u(strings)
+GROUP BY 1, 2
+```
+
+#### MVD
+
+```sql
+SELECT label, tags
+FROM "mvd_example"
+GROUP BY 1, 2
+```
+
+## How to ingest data as arrays
+
+You can ingest arrays in Druid as follows:
+
+* For native batch and streaming ingestion, configure the dimensions in [`dimensionsSpec`](../ingestion/ingestion-spec.md#dimensionsspec).
+Within `dimensionsSpec`, set `"useSchemaDiscovery": true`, and use `dimensions` to list the array inputs with type `auto`.  
+For an example, see [Ingesting arrays: Native batch and streaming ingestion](../querying/arrays.md#native-batch-and-streaming-ingestion).
+
+* For SQL-based batch ingestion, include the [query context parameter](../multi-stage-query/reference.md#context-parameters) `"arrayIngestMode": "array"` and reference the relevant array type (`VARCHAR ARRAY`, `BIGINT ARRAY`, or `DOUBLE ARRAY`) in the column descriptors.  
```

Comment:
"in the column descriptors" i'm not sure its clear what this means and I don't see anything else referring to it in the docs. Do you mean in the extern/extend type declaration thingy?

Ideally everyone should always be using that for any array data type from the external file, and then using `ARRAY_TO_MV` if storing as an mvd instead of array. I believe the web-console data loader already does this these days, regardless of arrayIngestMode, since the data is in fact arrays in the source files.

Reply:
"column descriptors" 😅  -- I got that phrase from https://github.com/apache/druid/blob/e9f723344b3de601b61f588fbbdbb4f14cd4b014/docs/multi-stage-query/reference.md?plain=1#L71

Reply:
updated the language and added the recommendation